File: LeetCode/python/ReverseString.py
class Solution(object):
    def reverseString(self, s):
        """
        :type s: str
        :rtype: str
        """
        sSize = len(s)
        
        if(sSize == 0 or sSize == 1):
            return s
            
        return s[::-1]
        
#测试程序
s = Solution()
print(s.reverseString(""))
print(s.reverseString("a"))
print(s.reverseString("abc"))
print(s.reverseString("abcd"))

File: LeetCode/iter1/c++/ReverseString.cpp
#include <iostream>
#include <string>

using namespace std;

string reverseString(string s) {
	if (s.size() == 0 || s.size() == 1)
		return s;

	int end = s.size() - 1;

	for (int i = 0;i<s.size()/2;i++)
	{
		char tmp = s[i];
		s[i] = s[end];
		s[end] = tmp;

		end--;
	}

	return s;
}

int main()
{
	cout << reverseString("") << endl
		<< reverseString("a") << endl
		<< reverseString("abc") << endl
		<< reverseString("abcd") << endl;

	return 0;
}